pkg/util/azureclient/roundtripper.go

```diff
@@ -0,0 +1,84 @@
+package azureclient
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+)
+
+const (
+	genevaOutboundRequestsTable = "outboundRequests"
+)
+
+func NewCustomRoundTripper(next http.RoundTripper) http.RoundTripper {
+	return &customRoundTripper{
+		next: next,
+	}
+}
+
+type customRoundTripper struct {
+	next http.RoundTripper
+}
+
+func (crt *customRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	startTime := time.Now()
+
+	correlationData := api.GetCorrelationDataFromCtx(req.Context())
+	if correlationData == nil {
+		correlationData = api.CreateCorrelationDataFromReq(req)
+	}
+
+	l := updateCorrelationDataAndEnrichLog(correlationData, utillog.GetLogger(), startTime)
+	l.Info("HttpRequestStart")
+
+	res, err := crt.next.RoundTrip(req)
+
+	l = updateCorrelationDataAndEnrichLogWithResponse(correlationData, l, res, startTime)
```

Comment:
I think I'm missing something here related to the RequestTime/StartTime.

1. Naming. Should we unify it to StartTime or RequestTime?
2. Why not just set the RequestTime when creating CorrelationData?